std/math.d

```diff
@@ -5600,17 +5583,7 @@ public:
             return true;
         else version(MIPS_Any)
             return true;
-        else version(AArch64)
-        {
-            auto oldState = getControlState();
-            // If exceptions are not supported, we set the bit but read it back as zero
-            // https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/aarch64/fpu/feenablxcpth.c
-            setControlState(oldState | (divByZeroException & allExceptions));
```

Comment:
Two notes on this: `setControlstate` is not pure, so I had to pull in the required assembly in ldc-developers/phobos#65 to get this to build. Also, I don't see the need for `allExceptions` here, `divByZeroException` is all that's going to come out of the logical and.

Reply:
Actually, as per the concession in the language spec, `setControlState()` can be pure also. As the FPControl itself saves state in `initialize()` and clears and restores state in `~this()`.

Reply:
True about the `AND` thing, depends how much you want to change in one go, considering the one-pr, one-change rule.

Reply:
These are all static member functions though, and I'd rather just clean up the unnecessary `&` here too.

Reply:
Ah... I was only looking at the public members.  The private static functions can be pure also.  We _know_ that the internal implementation doesn't have any side effects.

Reply:
It is possible to call the static member function `setControlState()` with non-exception bits, in which case it must reset them before exiting to be considered `pure`, which it can't do because it would defeat its purpose. However, it must have the `pure` attribute to be called from `hasExceptionTraps()`, so I can't call it there. As you see with my linked pull, my workaround is not to use `setControlState()`, but to just pull in the ARM32/64 assembly it uses, which I know is being used in a `pure` way internally to `hasExceptionTraps`, at least according to the spec.

Anyway, since the main Phobos branch doesn't implement the ARM32/64 paths anyway, this all just a placeholder for now, so it's fine. We can figure it out if and when we ever upstream the actual ARM code.